src/web/components/ClickToView.tsx

```diff
@@ -92,59 +106,59 @@ const Body = ({
 	width,
 }: {
 	children: React.ReactNode;
-	width: number;
+	width?: number;
 }) => (
 	<div
 		className={css`
-			${width > 300 ? textSans.small() : textSans.xsmall()}
+			${width && width > 300 ? textSans.small() : textSans.xsmall()}
 			a {
-				${width > 300 ? textSans.small() : textSans.xsmall()}
+				${width && width > 300 ? textSans.small() : textSans.xsmall()}
 			}
 		`}
 	>
 		{children}
 	</div>
 );
 
-export const ClickToView = ({ children, width, height, onAccept }: Props) => {
+export const ClickToView = ({
+	children,
+	width,
+	height,
+	overlayHeadline,
+	overlayBody,
+	overlayButtonText,
+	onAccept,
+}: Props) => {
 	const [showOverlay, setShowOverlay] = useState<boolean>(true);
 
 	const handleClick = () => {
 		setShowOverlay(false);
-		onAccept();
 	};
 
+	useEffect(() => {
+		if (!showOverlay) {
+			onAccept();
+		}
+	});
+
```

Comment:
I did this because for the specific use case of the UnsafeEmbedBlockComponent.

It requires that the embedIframe() function gets called after the associated iframe has been written into the DOM, this resizes the iframe.

Calling that method in the onClick handler doesnt work as the iframe has not been written into the dom at that point.

I need to do some more digging into this as there are still some issues and it doesn't work perfectly.

Reply:
You can seen this in action in the storybook:

https://5dfcbf3012392c0020e7140b-ymnejlgtya.chromatic.com/?path=/story/components-clicktoview--unsafe-embed-block-component-story

Reply:
Ah, I see. In this case you can wrap the call to the callback in a setTimeout, like `setTimeout(() => onMyThing())}` where the second param to setTimeout there is empty (so zero). This causes the execution to be put at the end of the js event stack.

Using useEffect is undesirably because it will fire the callback multiple times. You _couold_ use the custom hook we creted called `useOnce` to prevent this but the setTimeout is hopefully sufficient

Reply:
Great that works.